_posts/2018-12-17-til-umbrella-app-child-app-tests.md

```diff
@@ -0,0 +1,100 @@
+---
+author: Sophie DeBenedetto
+author_link: https://github.com/sophiedebenedetto
+categories: til
+date: 2018-12-17
+layout: post
+title:  TIL: How to Run Tests for One Child App in an Umbrella
+excerpt: >
+  Run all of the tests, or just a specific tests, for a given child app in an umbrella application with this handy command.
+---
+
+If you're working on an Elixir umbrella app with multiple children, then you know that running the tests for the entire umbrella app isn't always ideal. It could take a while to run and it can be hard to zero in on one particular set of failures when deving on just of the child apps.
+
+So, how can you run tests for just one specific child app?
+
+### Not Like This
+
+My first attempt to run just one child app's tests went a little something like this:
+
+```bash
+mix test test/github_client_test.exs
+==> learn_client
+Paths given to `mix test` did not match any directory/file: test/github_client_test.exs
+==> course_suite_client
+Paths given to `mix test` did not match any directory/file: test/github_client_test.exs
+==> github_client
+....
+
+Finished in 0.2 seconds
+4 tests, 0 failures
+
+Randomized with seed 395181
+==> deployer
+Paths given to `mix test` did not match any directory/file: test/github_client_test.exs
+==> deployer_web
+Paths given to `mix test` did not match any directory/file: test/github_client_test.exs
+```
+
+This _technically_ works--it _did_ run the test we specified--but its not quite what we want. Any `mix` command we run from the root of our umbrella app is recursively run from the root of each child app in the `apps/` directory. So, while this did run the `github_client_test.exs` test when the command was executed in the child app that contains that test, it also executed that command against *every other child app*. Resulting in these not-so-nice error messages:
+
+```bash
+Paths given to `mix test` did not match any directory/file:
+test/github_client_test.exs
+```
+
+Not ideal.
+
+### Like This
+To run _all_ of the tests for just one specific child app, we can run the following from the root of the umbrella app:
+
+```elixir
+mix cmd --app child_app_name mix test --color
+```
+
+We use `mix cmd --app` to specific the app within which we want to run a given `mix` command. Then we specify the `mix test` command. Additionally, we can specify a test file and even line number to run:
```

Comment:
to specify.
Do you want to change words maybe, as you used a little too many times the word specify :)